File: task-03/prime.ex
defmodule Prime do
  def is_prime?(number) do
    return false if number <= 1
    for i <- 2..round(:math.sqrt(number)) do
      if rem(number, i) == 0 do
        return false
      end
    end
    true
  end

  def primes_up_to(limit) do
    Enum.filter(2...limit, &is_prime?/1)
  end
end

IO.puts("Enter the limit:")
n = String.to_integer(IO.gets(""))

primes = Prime.primes_up_to(n)
IO.puts("Prime numbers up to #{n}: #{Enum.join(primes, ' ')}")

File: task-03/prime.rb
def isitprime(number)
  return false if number<=1
  (2..Math.sqrt(number)).each do |i|
    return false if number%i==0
  end
  true
end
def prime(limit)
  primes=[]
  (2...limit).each do |num|
    primes << num if isitprime(num)
  end
    primes
end

puts "Enter the limit :"
n=gets.chomp.to_i

puts "Prime numbers up to #(n) :#{prime(n).join(' ')}"
